feature_engine/transformation/log.py

```diff
@@ -164,3 +164,205 @@ def _more_tags(self):
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features].
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y: pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+
+        Raises
+        ------
+        TypeError
+            - If the input is not a Pandas DataFrame
+            - If any of the user provided variables are not numerical
+        ValueError
+            - If there are no numerical variables in the df or the df is empty
+            - If the variable(s) contain null values
+            - If some variables contain zero or negative values after adding constant C
+
+        Returns
+        -------
+        self
+        """
+
+        # check input dataframe
+        X = super().fit(X)
+
+        self.C_ = self.C
+
+        # calculate C to add to each variable
+        if self.C == "auto":
+            self.C_ = dict(X[self.variables_].min(axis=0).abs() + 1)
+
+        # check contains zero or negative values
+        if (X[self.variables_] + self.C_ <= 0).any().any():
+            raise ValueError(
+                "Some variables contain zero or negative values after adding"
+                + "constant C, can't apply log"
+            )
+
+        self.n_features_in_ = X.shape[1]
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transforms the variables with the logarithm of x plus a constant C.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features]
+            The data to be transformed.
+
+        Raises
+        ------
+        TypeError
+            If the input is not a Pandas DataFrame
+        ValueError
+            - If the variable(s) contain null values
+            - If the df has different number of features than the df used in fit()
+            - If some variables contains zero or negative values after adding constant C
+
+        Returns
+        -------
+        X: pandas dataframe
+            The dataframe with the transformed variables.
+        """
+
+        # check input dataframe and if class was fitted
+        X = super().transform(X)
+
+        # check contains zero or negative values
+        if (X[self.variables_] + self.C_ <= 0).any().any():
+            raise ValueError(
+                "Some variables contain zero or negative values after adding"
+                + "constant C, can't apply log"
+            )
+
+        # transform
+        if self.base == "e":
+            X.loc[:, self.variables_] = np.log(X.loc[:, self.variables_] + self.C_)
+        elif self.base == "10":
+            X.loc[:, self.variables_] = np.log10(X.loc[:, self.variables_] + self.C_)
+
+        return X
+
+    def inverse_transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transforms the variables using the exponential transformation.
```

Comment:
The method should read "Convert the data back to the original representation." as per sklearn convention for inverse_transform.

Reply:
got it